runtime/pallene_core.h

```diff
@@ -226,4 +230,46 @@ int pallene_is_truthy(const TValue *v)
     return !(ttisnil(v) || (ttisboolean(v) && bvalue(v) == 0));
 }
 
+static inline
+Table *pallene_new_table(lua_State *L, lua_Integer n)
+{
+    Table *t = luaH_new(L);
+    if (n > 0) {
+        luaH_resize(L, t, 0, n);
+    }
+    return t;
+}
+
+static const TValue PALLENE_ABSENTKEY = {ABSTKEYCONSTANT};
+
+/* This function is a speciallization of luaH_getshortstr from ltable.c */
+static inline
+TValue *pallene_getstr(Table *t, TString *key, size_t *pos)
+{
+    if (PALLENE_LIKELY(*pos < sizenode(t))) {
+       Node *n = gnode(t, *pos);
+       if (PALLENE_LIKELY(keyisshrstr(n) && eqshrstr(keystrval(n), key)))
+           return gval(n);
+    }
```

Comment:
For `gcc` the LIKELY macro is interpreted to assume that the condition is true at least 90% of the time. In this case, using LIKELY for this cache signifies that we expect that the cache will be a hit at least 81% of the time.

Is it safe to assume that? If not, it would be more accurate to use a regular if-statement without the PALLENE_LIKELY annotations.

Reply:
That really depends on the benchmark that you are doing. In fact, any simple benchmark like ours will benefit this LIKELY case because it will always hit the cache eventually. Bigger programs might have an issue. I will remove the LIKELY for now to be more fair.